stubs/pluggy/pluggy/_manager.pyi

```diff
@@ -0,0 +1,65 @@
+import sys
+from _typeshed import Incomplete
+from collections.abc import Callable, Iterable, Mapping, Sequence
+from typing import Any
+from typing_extensions import Final, TypeAlias
+
+from ._hooks import (
+    HookImpl as HookImpl,
+    HookSpec as HookSpec,
+    _HookCaller,
+    _HookImplOpts,
+    _HookRelay,
+    _HookSpecOpts,
+    _Namespace,
+    _Plugin,
+    normalize_hookimpl_opts as normalize_hookimpl_opts,
+)
+from ._result import _Result as _Result
+from ._tracing import TagTracerSub
+
+if sys.version_info >= (3, 8):
+    from importlib.metadata import Distribution
+else:
+    Distribution: TypeAlias = Any
```

Comment:
We actually allow our stubs packages to have non-stubs packages as dependencies these days. So you could potentially add `importlib_metadata` as a dependency to this stubs package on Python 3.7. But also, Python 3.7 is now end-of-life and we'll slowly be phasing support out from typeshed over the coming months (see #10113). So maybe it's not worth the bother. I'm easy either way; up to you.

Reply:
@AlexWaygood I think something is still slipping away from me here - if I apply this patch
```diff
--- a/stubs/pluggy/METADATA.toml
+++ b/stubs/pluggy/METADATA.toml
@@ -1 +1,2 @@
 version = "1.2.0"
+requires = ["importlib-metadata;python_version<'3.8'"]
--- a/stubs/pluggy/pluggy/_manager.pyi
+++ b/stubs/pluggy/pluggy/_manager.pyi
@@ -21,7 +21,7 @@ from ._tracing import TagTracerSub
 if sys.version_info >= (3, 8):
     from importlib.metadata import Distribution
 else:
-    Distribution: TypeAlias = Any
+    from importlib_metadata import Distribution

 _BeforeTrace: TypeAlias = Callable[[str, Sequence[HookImpl], Mapping[str, Any]], None]
 _AfterTrace: TypeAlias = Callable[[_Result[Any], str, Sequence[HookImpl], Mapping[str, Any]], None]
 ```
then `mypy` doesn't like it, since the requirement is not available:
```sh
$ python tests/mypy_test.py --platform linux --python-version=3.7 stubs/pluggy
*** Testing Python 3.7 on linux
Testing third-party packages...
testing pluggy (5 files)... failure

stubs/pluggy/pluggy/_manager.pyi:24: error: Cannot find implementation or library stub for module named "importlib_metadata"  [import]
stubs/pluggy/pluggy/_manager.pyi:24: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports


--- exit status 1, 5 files checked ---
```

Reply:
Hrm, I'll try to take a look today

Reply:
> then `mypy` doesn't like it, since the requirement is not available:
> 
> ```shell
> $ python tests/mypy_test.py --platform linux --python-version=3.7 stubs/pluggy
> *** Testing Python 3.7 on linux
> Testing third-party packages...
> testing pluggy (5 files)... failure
> 
> stubs/pluggy/pluggy/_manager.pyi:24: error: Cannot find implementation or library stub for module named "importlib_metadata"  [import]
> stubs/pluggy/pluggy/_manager.pyi:24: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
> 
> 
> --- exit status 1, 5 files checked ---
> ```

Oh. You probably need to actually run the test _using_ Python 3.7 for the requirement to be installed by mypy_test.py. The --python-version 3.7 flag probably isn't enough.

(mypy_test.py these days dynamically creates a virtual environment on the fly for each package that has non-types dependencies, and uses pip to install all non-types dependencies into the venv. But pip is probably just ignoring the importlib-metadata dependency because you're python version is >=3.8? At a guess?)

Reply:
@AlexWaygood ahh, indeed! Sorry for the noise 🥲

Reply:
Okay, looks like adding `importlib_metadata` as a dependency on 3.7 works with `mypy_test.py`, but _does_ break just about every other test we have. It's all probably _fixable_, but some of it would be a bit of a pain to fix, so maybe it would be better to just have `Distribution: TypeAlias = Any`, the same way you had it originally 🙃

Sorry for the churn!!

Reply:
@AlexWaygood yeah I mean since py3.7 is about to be dropped anyway, it doesn't matter this much anymore 😎

Reply:
@AlexWaygood looks like [this job](https://github.com/python/typeshed/actions/runs/5610198580/jobs/10264878450?pr=10473) failed now because of GH hiccups, not because of code changes.

Reply:
> @AlexWaygood looks like [this job](https://github.com/python/typeshed/actions/runs/5610198580/jobs/10264878450?pr=10473) failed now because of GH hiccups, not because of code changes.

Yeah, it's been really flaky recently. Not sure why; I should probably open a bug report...